File: Atcoder_#345/ruby/a_leftrightarrow.rb
# ユーザーからの入力を取得
s = gets.strip

# 文字列が '<' で始まるかどうかを確認するメソッド
def first_str?(s)
  s[0] == "<"
end

# 文字列内に '<' または '>' が含まれているかどうかを確認するメソッド
def during_str?(s)
  !s[1..-2].include?("<") && !s[1..-2].include?(">")
end

# 文字列が '>' で終わるかどうかを確認するメソッド
def last_str?(s)
  s[-1] == ">"
end

# 全ての条件を満たしているかどうかを確認するメソッド
def check_string(s)
  if first_str?(s) && during_str?(s) && last_str?(s)
    puts "Yes"
  else
    puts "No"
  end
end

# 文字列をチェック
check_string(s)

File: Atcoder_#345/javascript/a_leftrightarrow.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let inputLines = [];
rl.on('line', (line) => {
  inputLines.push(line);
}).on('close', () => {
  processInput(inputLines);
});

function processInput(input) {
  const s = input[0];

  function firstStr(s) {
    return s[0] === "<";
  }

  function duringStr(s) {
    return !s.slice(1, -1).includes("<") && !s.slice(1, -1).includes(">");
  }

  function lastStr(s) {
    return s[s.length - 1] === ">";
  }

  function checkString(s) {
    if (firstStr(s) && duringStr(s) && lastStr(s)) {
      console.log("Yes");
    } else {
      console.log("No");
    }
  }

  checkString(s);  // 文字列の検証を実行
}
